src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -152,129 +180,92 @@ public void initForViewByGiverQuestionRecipient(InstructorAttributes instructor,
-        
-        Map<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> sortedResponses 
-                     = bundle.getResponsesSortedByRecipientQuestionGiver(true);
-       
-        buildSectionPanelsForRecipientQuestionGiver(sortedResponses);
     }
     
-    /**
-     * Constructs section panels for every section.
-     * 
-     * TODO reduce code duplication across the build.*Panels.* methods 
-     * @param sortedResponses
-     */
-    private void buildSectionPanelsForGiverQuestionRecipient(
-                                    Map<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> sortedResponses) {
-        viewType = ViewType.GIVER_QUESTION_RECIPIENT;
-        final String additionalInfoId = "giver-%s-question-%s";
-        
-        LinkedHashMap<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> responsesGroupedByTeam 
-                = bundle.getQuestionResponseMapByGiverTeam();
-        
-        
+    private void buildSectionPanelsForViewByParticipantParticipantQuestion(
+                         Map<String, Map<String, List<FeedbackResponseAttributes>>> sortedResponses,
+                         String additionalInfoId) {
         sectionPanels = new LinkedHashMap<String, InstructorFeedbackResultsSectionPanel>();
         InstructorFeedbackResultsSectionPanel sectionPanel = new InstructorFeedbackResultsSectionPanel();
         
-        // Maintain previous section and previous team while iterating through the loop
+        // Maintain previous section and previous team while iterating through the recipients
         // initialize the previous section to "None"
         String prevSection = Const.DEFAULT_SECTION;
         String prevTeam = "";
         
-        // Used for displaying sections without responses for the course at the end
         Set<String> sectionsWithResponses = new HashSet<String>();
-        // Used for displaying teams without responses for every section
-        LinkedHashSet<String> teamsWithResponses = new LinkedHashSet<String>();
-        // Used for displaying participants without responses for every team
-        Set<String> teamMembersWithResponses = new HashSet<String>();                                
-        
-        int giverIndex = this.startIndex;
-        // Iterates through the response givers
-        for (Map.Entry<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> responsesFromGiver : 
-                                                                    sortedResponses.entrySet()) {
-            giverIndex += 1;
-            String giverIdentifier = responsesFromGiver.getKey();
+        Set<String> teamsWithResponses = new HashSet<String>();
+        Set<String> teamMembersWithResponses = new HashSet<String>();      
+          
+        // Iterate through the primary participant
+        int recipientIndex = this.getStartIndex();
```

Comment:
Not too important, but I don't think there's any need for calling the method instead of accessing the variable directly?
There are getters and setters for class-scope variables used throughout the file.